### ISSUE TITLE
[E-2] [39] Add baseline lab linked to EDA findings

### ISSUE THREAD

**Issue by wildersa** [wildersa/ml-starter-lab-kit#74]
# [39] Add baseline lab linked to EDA findings — compare simple baselines before real experiments

## Dependency

Depends on #73.

Do not start this issue until #73 is merged. If #73 is not merged, stop and state that dependency in the PR body.

## Agent behavior reinforcement

You are an experienced Python ML education tooling engineer with deep expertise in scikit-learn baselines, EDA-driven experiment setup, generated project templates, MLflow-optional tracking, and regression tests for generated project behavior.

Focus on a small educational baseline lab that requires EDA and writes comparison artifacts. Do not solve this by building AutoML, hyperparameter optimization, deep learning support, advanced model selection, or changing the main train/evaluate pipeline semantics.

## Problem

Guided Learning should help the user understand why a baseline matters before running real experiments. The baseline lab must be tied to EDA artifacts and remain educational, not AutoML.

Expected behavior:

```txt
python -m <package>.lab baseline

requires configs/eda_summary.json and creates:
- configs/baseline_results.json
- reports/baseline-results.md
```

## Explicit design decisions

| Decision | Required behavior |
|---|---|
| Supported inputs/signature | Add baseline command through `lab.py`; optional direct wrapper allowed if consistent with repo style. |
| Source of truth | `configs/config.json` and `configs/eda_summary.json`. |
| Precedence rules | No EDA artifact means baseline lab is blocked. |
| Fallback behavior | If task/model type is unsupported, write a clear unsupported message and no misleading metrics. |
| Safe output contract | Save metrics/results, not raw full dataset. |
| Unsupported behavior | No AutoML, no hyperparameter search, no deep learning, no production model replacement. |

## P0 objectives

- [ ] P0.1 — Baseline lab is blocked when `configs/eda_summary.json` is missing.
- [ ] P0.2 — With EDA and dataset present, baseline lab creates `configs/baseline_results.json` and `reports/baseline-results.md`.
- [ ] P0.3 — Classification/regression baselines use simple scikit-learn baseline models and beginner-friendly metrics.
- [ ] P0.4 — Streamlit workspace can show baseline status/results or guide the user to run baseline.
- [ ] P0.5 — Baseline lab does not replace or rewrite the main train/evaluate pipeline behavior.

## Author-defined TDD contract

| P0 | Test file | Test intent | Setup required | Required assertion | Regression this catches |
|---|---|---|---|---|---|
| P0.1 | `tests/test_output_logic.py` | baseline blocked without EDA | Generate guided project with dataset but no `eda_summary.json` | baseline command/function prints EDA prerequisite and creates no baseline artifacts | baseline runs before EDA |
| P0.2 | `tests/test_output_logic.py` | baseline artifacts generated | Provide CSV, config, and `eda_summary.json` | both baseline result files exist | no persistent baseline output |
| P0.3 | `tests/test_output_logic.py` | supervised baseline metrics | Use small classification or regression fixture | JSON contains baseline model name and at least one relevant metric | baseline result is empty/generic |
| P0.4 | `tests/test_templates.py` or `tests/test_generator.py` | workspace baseline section | Inspect workspace | Baseline Lab section references baseline results/prerequisite | UI cannot surface baseline |
| P0.5 | `tests/test_generator.py` | train/evaluate preserved | Generate project | `train.py` and `evaluate.py` still exist and are not replaced by baseline lab | baseline hijacks main pipeline |

## Files to inspect

- `templates/project/package/lab.py.tpl`
- `templates/project/package/learning_workspace.py.tpl`
- generated core layer from prior issues
- `templates/project/package/train.py.tpl`
- `templates/project/package/evaluate.py.tpl`
- `templates/env/requirements-ml.txt.tpl`
- `tests/test_output_logic.py`
- `tests/test_generator.py`

## Expected files or areas to change

- new generated baseline core/template module, e.g. `baseline_lab.py.tpl` or core operation
- `templates/project/package/lab.py.tpl`
- `templates/project/package/learning_workspace.py.tpl`
- tests for baseline artifacts and gating

## Files not allowed to change

Do not change these unless required by P0:

- main train/evaluate algorithms except imports/wrappers needed for compatibility
- advisor recommendation heuristics
- demo dataset generation
- notebook templates

## Non-solutions

- AutoML.
- Hyperparameter optimization.
- Training many models automatically.
- Replacing `train.py` with baseline lab.
- Running baseline before EDA.
- Reporting metrics without dataset/config grounding.

## Validation commands

```bash
python -m unittest tests.test_output_logic tests.test_generator tests.test_templates
```

## PR description must include

- Baseline behavior per supported task.
- Artifact schema summary.
- P0 coverage table with tests/assertions.
- Confirmation that main train/evaluate flow was not replaced.

## Acceptance checklist

- [ ] All P0 objectives satisfied.
- [ ] TDD contract covered.
- [ ] Baseline requires EDA.
- [ ] Results are persisted as JSON and Markdown.
- [ ] No AutoML/hyperparameter search added.
- [ ] PR links this issue with `Fixes #...` or `Closes #...`.
